percona/controller/pgcluster/controller_test.go

```
@@ -1622,3 +1622,118 @@ var _ = Describe("Validate TLS", Ordered, func() {
 		checkSecretProjectionWithCA(cr, cr.Spec.Secrets.CustomReplicationClientTLSSecret, secretName)
 	})
 })
+
+type saTestClient struct {
+	client.Client
+
+	crName string
+	ns     string
+}
+
+func (sc *saTestClient) checkObject(ctx context.Context, obj client.Object) error {
+	sts, ok := obj.(*appsv1.StatefulSet)
+	if !ok {
+		return nil
```

should we return an error in that case?